docs/markdown/snippets/prelink.md

```diff
@@ -0,0 +1,5 @@
+## Add support for prelinked static libraries
+
+The static library gains a new `prelink` keyword argument that can be
+used to prelink object files in that target. This is mostly useful in
+embedded projects.
```

Comment:
I think partial linking is less rare in embedded projects _only_ because static libraries are less rare in embedded projects. I would not mention "embedded" here. This new feature is only about static libraries anyway.

I found the two use cases listed in #5499 very interesting, I think it would be much better to mention these: 
- namespaces more or less equivalent to `dllexport` but for static libraries 
- partial LTO which is better than no LTO at all when building with different toolchains.